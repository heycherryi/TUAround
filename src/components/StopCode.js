import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './StopCode.scss';

function StopCode({ stop }) {
  return (
    <span className="stop-code badge bg-white text-black border border-secondary"><Link to={`/stops/${stop.stop_code}`}>{stop.stop_code}</Link></span>
  );
}

StopCode.propTypes = {
  stop: PropTypes.object.isRequired,
};

export default StopCode;
