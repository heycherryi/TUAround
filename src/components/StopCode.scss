.stop-code {
  a:link, a:hover, a:active, a:visited {
    color: black;
    text-decoration: none;
  }
}
