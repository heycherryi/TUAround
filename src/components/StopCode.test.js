/* globals test, expect */

import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import StopCode from './StopCode';
import stopFixture from '../fixtures/stop-10AHERNN.json';

test('renders StopCode', () => {
  const { container } = render(
    <Router>
      <StopCode stop={stopFixture} />
    </Router>,
  );
  expect(screen.getByText('10AHERNN')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
