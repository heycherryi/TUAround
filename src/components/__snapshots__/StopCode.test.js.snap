// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StopCode 1`] = `
<div>
  <span
    class="stop-code badge bg-white text-black border border-secondary"
  >
    <a
      href="/stops/10AHERNN"
    >
      10AHERNN
    </a>
  </span>
</div>
`;
